master/buildbot/reporters/utils.py

```diff
@@ -30,7 +30,7 @@ def getPreviousBuild(master, build):
     n = build['number'] - 1
     while n >= 0:
         prev = yield master.data.get(("builders", build['builderid'], "builds", n))
-        if prev['results'] != RETRY:
+        if prev and prev['results'] != RETRY:
```

Comment:
if n > 0 then prev should always be defined. What is your value of n? did you remove data in your database somehow?

The double check is okay though